pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -269,81 +273,122 @@ func init() {
+		k.cancel = cancel
+		builder.WithContext(ctx)
 
-	k.processLabelJoins()
-	k.instance.LabelsAsTags = mergeLabelsOrAnnotationAsTags(metadataAsTags.GetResourcesLabelsAsTags(), k.instance.LabelsAsTags)
-	k.processLabelsAsTags()
+		resyncPeriod := k.instance.ResyncPeriod
+		if resyncPeriod == 0 {
+			resyncPeriod = pkgconfigsetup.Datadog().GetInt("kubernetes_informers_resync_period")
+		}
 
-	// We need to merge the user-defined annotations as tags with the default annotations first
-	mergedAnnotationsAsTags := mergeLabelsOrAnnotationAsTags(metadataAsTags.GetResourcesAnnotationsAsTags(), defaultAnnotationsAsTags())
-	k.instance.AnnotationsAsTags = mergeLabelsOrAnnotationAsTags(mergedAnnotationsAsTags, k.instance.AnnotationsAsTags)
-	k.processAnnotationsAsTags()
+		builder.WithResync(time.Duration(resyncPeriod) * time.Second)
 
-	// Prepare labels mapper
-	k.mergeLabelsMapper(defaultLabelsMapper())
+		builder.WithGenerateStoresFunc(builder.GenerateStores)
+
+		// configure custom resources required for extended features and
+		// compatibility across deprecated/removed versions of APIs
+		builder.WithGenerateCustomResourceStoresFunc(builder.GenerateCustomResourceStoresFunc)
+		builder.WithCustomResourceStoreFactories(custom.factories...)
+		builder.WithCustomResourceClients(custom.clients)
+
+		// Enable exposing resource annotations explicitly for kube_<resource>_annotations metadata metrics.
+		// Equivalent to configuring --metric-annotations-allowlist.
+		allowedAnnotations := map[string][]string{}
+		for _, collector := range collectors {
+			// Any annotation can be used for label joins.
+			allowedAnnotations[collector] = []string{"*"}
+		}
 
-	builder := kubestatemetrics.New()
-	builder.WithUsingAPIServerCache(k.instance.UseAPIServerCache)
+		builder.WithAllowAnnotations(allowedAnnotations)
 
-	k.configurePodCollection(builder, k.instance.Collectors)
+		// Enable exposing resource labels explicitly for kube_<resource>_labels metadata metrics.
+		// Equivalent to configuring --metric-labels-allowlist.
+		allowedLabels := map[string][]string{}
+		for _, collector := range collectors {
+			// Any label can be used for label joins.
+			allowedLabels[collector] = []string{"*"}
+		}
+
+		if err = builder.WithAllowLabels(allowedLabels); err != nil {
+			configureErr = err
+		}
+
+		if err := builder.WithEnabledResources(collectors); err != nil {
+			configureErr = err
+		}
+
+		// Start the collection process
+		k.allStores = builder.BuildStores()
+
+	})
+	return configureErr
+}
+
+// configureCheck retries _configure() if it previously failed
+func (k *KSMCheck) configureCheck() error {
+	if err := k.configureRetrier.TriggerRetry(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// _configureCheck()
+// 1. executes the configuration steps dependent on coordination with the API Server that can safely be retried
+// 2. executes the KSM Builder steps as a singleton
+func _configureCheck(k *KSMCheck) error {
```

Comment:
We usually don't `_` functions. You can simply put:
```
	if err := k.configureRetrier.TriggerRetry(); err != nil {
		return err
	}
	return nil
```

in `Run` and call the function `configureCheck` and attach it to `k`. It would be a `func() error` and could be passed directly to Retrier.

Also as I commented in previous review, we don't want to return this error, but the `LastTryError`